### ISSUE TITLE
Unable to config SMTP with OVH

### ISSUE THREAD

**Issue by Mr-KayJayDee** [knadh/listmonk#1366]
**Version:**
 - listmonk :v2.4.0
 - OS: Debian 11

**Description of the bug and steps to reproduce:**
I am trying to configure the SMTP settings of the app with my ovh credentials. I changed some settings and tried many times and i cannot get an email on my test box. Sometimes i even get a "error sending SMTP test (Test connection): context deadline exceeded" depending on the setting i changed

Note that sometimes i get a success popup but i receive nothing

**Screenshots:**
![image](https://github.com/knadh/listmonk/assets/46600662/8b8bdca4-34d8-499d-8629-814db5d80bbb)

As you can see on the next picture my settings are correct: 

![image](https://github.com/knadh/listmonk/assets/46600662/e8c03f08-095f-4b87-a920-960b532a40cd)

**Comment by knadh** [comment]
Under the `Send e-mail` button for testing SMTP connection, the log box should show any errors in read. Does it show anything at all? If no, then the SMTP server accepted the e-mail without returning any errors but silently discarded it.

**Comment by Mr-KayJayDee** [comment]
Sometimes it shows this error: "context deadline exceeded" the same as in the logs page, no other error

**Comment by knadh** [comment]
That happens where there is a timeout, where listmonk isn't being able to reach the target SMTP server. Looks like something's blocking the connections. Could be firewall, Docker config if you're using Docker. Might want to use something like telnet to see if you can establish a connection to the environment.

**Comment by Mr-KayJayDee** [comment]
I'm using binary, there are no firewall rules blocking smtp, i tried using a small nodemailer code with nodejs and the same credentials on the same server and the mail went through.

**Comment by knadh** [comment]
I'm at a loss here. Could it be something SELinux that's blocking the listmonk binary from opening that connection? @mr-karan ?

**Comment by Mr-KayJayDee** [comment]
> @Mr-KayJayDee Probably the upstream SMTP server is taking more than 5s to respond. Can you try `openssl s_client -connect ssl0.ovh.net:25 -starttls smtp` and see how long it takes before you see `250` response from the server? And if it's several few seconds, then you can increase `wait_timeout` config in UI from 5s to probably a higher value like 30s or something?

The answer was instant here is the output:

```bash
CONNECTED(00000003)
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
verify return:1
depth=0 CN = ns0.ovh.net
verify return:1
---
Certificate chain
 0 s:CN = ns0.ovh.net
   i:C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
 1 s:C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
   i:C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIHPjCCBiagAwIBAgIQWaLZM+2tD93bROHix3v2AjANBgkqhkiG9w0BAQsFADCB
jzELMAkGA1UEBhMCR0IxGzAZBgNVBAgTEkdyZWF0ZXIgTWFuY2hlc3RlcjEQMA4G
A1UEBxMHU2FsZm9yZDEYMBYGA1UEChMPU2VjdGlnbyBMaW1pdGVkMTcwNQYDVQQD
Ey5TZWN0aWdvIFJTQSBEb21haW4gVmFsaWRhdGlvbiBTZWN1cmUgU2VydmVyIENB
MB4XDTIyMDgzMTAwMDAwMFoXDTIzMDgzMTIzNTk1OVowFjEUMBIGA1UEAxMLbnMw
Lm92aC5uZXQwggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDFjLm7kVkc
bKqp70xdzdjQZa6xIP/BFj5kZJU2WlVo37WagaaqHoPhc9JCa2Qxl40qGdwUBjp/
mVuW7XlZCpx/xAC/33kKEFj5Tbnto+enA/1gIh5PX84bFNkHK8YTm6taVPjhSrKQ
TyRtCW/IqZoxPBas8F+f/EBWYA/XdKqdV02Q4BLyN7giBp3RfbFL3ixh9erBjSeH
iI+ujGTrZ47FYfUaXVmLL/2C3GD8HO4mdwphDwtBuuS4GRUNtJyvn/da0xO9BNO5
TnEpelggVgp2eX6EAyeVMVFoeYB1jpTawtPz1KRsPl6d+IYnL1xi84F8NeExyXkR
ZbuNRZBkkL+yZHXRQNTcIcFUPOMYqH6IQiqcj6v24ys9Y68UZR+Lh8Vcxv2DVu3b
MC44Fc92LbahaSXoVIQEDYVv6XOD3o6nZ/8xbGlExGp7GFt5+K7+VY4LmbMatZh/
ikdYs2wsN5iNUCUlo8Fbt4NjujvL+G021597IgxYvOepHkL2dhgeYLsMLQmvupNZ
6Km8PBB1bAvjeuJV1MN2s87+HVw28Q+ygL3hmqXg5GIJB9c98UoncFtTEaiSiI0j
wzHwqzrhdzAI/wOZMmk+Une7MtRBen3lknWGkjasjdBE3GEWtIvgvqFPtCAyMusz
xgCc4r0OrigPE9ZAmde8abSWPbtah8QJlQIDAQABo4IDDDCCAwgwHwYDVR0jBBgw
FoAUjYxexFStiuF36Zv5mwXhuAGNYeEwHQYDVR0OBBYEFGAFI5qoSWcQIG69pyw8
kAxd+REcMA4GA1UdDwEB/wQEAwIFoDAMBgNVHRMBAf8EAjAAMB0GA1UdJQQWMBQG
CCsGAQUFBwMBBggrBgEFBQcDAjBJBgNVHSAEQjBAMDQGCysGAQQBsjEBAgIHMCUw
IwYIKwYBBQUHAgEWF2h0dHBzOi8vc2VjdGlnby5jb20vQ1BTMAgGBmeBDAECATCB
hAYIKwYBBQUHAQEEeDB2ME8GCCsGAQUFBzAChkNodHRwOi8vY3J0LnNlY3RpZ28u
Y29tL1NlY3RpZ29SU0FEb21haW5WYWxpZGF0aW9uU2VjdXJlU2VydmVyQ0EuY3J0
MCMGCCsGAQUFBzABhhdodHRwOi8vb2NzcC5zZWN0aWdvLmNvbTCCAXwGCisGAQQB
1nkCBAIEggFsBIIBaAFmAHUArfe++nz/EMiLnT2cHj4YarRnKV3PsQwkyoWGNOvc
gooAAAGC80tXEwAABAMARjBEAiAZeG61NzuO+J7ZeeSVRCtOq61M2O624VKwgu1X
utbhEQIgCew/ecq0UqBxnbq/V2ZKiZsoCAh09ErN4WqXQh5Ar30AdgB6MoxU2Lct
tiDqOOBSHumEFnAyE4VNO9IrwTpXo1LrUgAAAYLzS1clAAAEAwBHMEUCIAd06G1V
epz3wC2w+bcVf2S5oj0F4P4fKwNl0/nj6oxoAiEAo8J+7qWpA2GvKgs0JjRWrF/C
6DXnI/hBLqghIPSwW1oAdQDoPtDaPvUGNTLnVyi8iWvJA9PL0RFr7Otp4Xd9bQa9
bgAAAYLzS1b3AAAEAwBGMEQCIAh46SfDlyjk020Ea63A7gwmQKHdJG5O4EU4pfU2
fsU1AiBVhY1C37j23Q1OsAFl8p7Y7Se2JEW0gCpagK4IKpFkMjA3BgNVHREEMDAu
ggtuczAub3ZoLm5ldIIRc210cC5tYWlsLm92aC5uZXSCDHNzbDAub3ZoLm5ldDAN
BgkqhkiG9w0BAQsFAAOCAQEAja5teuJ7aNYQUfEllaQHL8zqlppxlz2uGVagtW+3
rHvPdYzmVMBaHndpJo1eVxrhEm4fibY0V3SM8MnXNNa5J1gHbGCdEQCtblA0d+/l
y4wErohQamopC+T+PWngyPhHY3v1SLSxUXAMR+bnAZ04tU7DOQU6Bn6aN0Jvlw7j
bIZlnU7NXKcZrBkiXRhc3n30FVAzKMoii+MGf6ZjolG/hVQPMvuTZiow/6hIDfG+
sSQkgZSf3xao+6Ln9rLwp16mx83+sFDGksojMAk6OioWN8v2pZNZHZROvEd6QSXS
646d9hc3jEOxOn+7I1kD4t9gxvqMHfbiyCDu/vPqx6X42w==
-----END CERTIFICATE-----
subject=CN = ns0.ovh.net

issuer=C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA

---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA
Server Temp Key: ECDH, P-384, 384 bits
---
SSL handshake has read 4411 bytes and written 495 bytes
Verification: OK
---
New, TLSv1.2, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 4096 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID: 5E11000058B43FB6E4822E1C6DFB278AB9068894DE22263467A7F554D290CF24
    Session-ID-ctx: 
    Master-Key: 17B5F5BF5BC846E4A9187C7B12AD44F44AF10BE7854871CC96997F9D12A82EF5A442797CFB4B00ABB461938F43944C86
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1686826934
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: yes
---
250 8BITMIME
```

**Comment by Mr-KayJayDee** [comment]
Note: after 1min or more i get this:

```
421 Service not available, closing transmission channel
221  Service closing transmission channel
```

**Comment by mr-karan** [comment]
@Mr-KayJayDee I'm unsure what is happening here. Like @knadh suggested, please check if there are any application permissions required to open ports for outbound connections?

**Comment by Mr-KayJayDee** [comment]
> @Mr-KayJayDee I'm unsure what is happening here. Like @knadh suggested, please check if there are any application permissions required to open ports for outbound connections? 

I'm running the app as root in the terminal for now, does running it this way causes issues ?

**Comment by knadh** [comment]
> I'm running the app as root in the terminal for now, does running it this way causes issues ?

This shouldn't matter.

**Comment by Mr-KayJayDee** [comment]
hmm, and does context deadline exceeded tells you something ?

**Comment by Mr-KayJayDee** [comment]
```sh
root@bot:/home/bots/mrkayjaydee# telnet ssl0.ovh.net 25
Trying 193.70.18.144...
Connected to ssl0.ovh.net.
Escape character is '^]'.
220 GARM-97G002 Friday, June 16, 2023
```

**Comment by Mr-KayJayDee** [comment]
Using gmail smtp it does work

**Comment by Mr-KayJayDee** [comment]
Is there a debug mode to see all actions and error, or even a log file ?

**Comment by Mr-KayJayDee** [comment]
Just tried hosting it with railway and have the same issue

**Comment by Mr-KayJayDee** [comment]
Here is the result of the request made by the test button and then displaying success but i get no email

{
	"data": [
		"2023/06/18 16:53:53 main.go:99: v2.4.0 (c668523 2023-03-20T13:50:31Z)",
		"2023/06/18 16:53:53 init.go:138: reading config: config.toml",
		"2023/06/18 16:53:53 init.go:266: connecting to db: listmonk_db:5432/listmonk",
		"2023/06/18 16:53:55 init.go:585: media upload provider: filesystem",
		"2023/06/18 16:53:55 init.go:509: loaded email (SMTP) messenger: contact@mrkayjaydee.xyz@ssl0.ovh.net",
		"2023/06/18 16:55:37 settings.go:260: error sending SMTP test (Test connection): context deadline exceeded"
	]
}

**Comment by Mr-KayJayDee** [comment]
Hi ! Any news on this ?

**Comment by knadh** [comment]
Closing this issue. @Mr-KayJayDee apologies for not being able to find a solution. The connection issue seems to be environment + host dependent, which we're unable to reproduce elsewhere.

**Comment by MaximilianKohler** [comment]
TLDR: seemed to be a server issue for me.

---

Interesting. I've been using listmonk with Amazon SES for over a year now and just started experiencing this `context deadline exceeded` error. It seems to be happening intermittently.

It started right after I updated my server's nginx version, so I tried rebooting the server and that changed the error to `SMTP AUTH extension not found`.

A search turns up these 3 related issues but no fix:
* https://github.com/knadh/listmonk/issues/1492
* https://github.com/knadh/listmonk/issues/629
* https://github.com/knadh/listmonk/issues/575

I went to the `settings -> SMTP` page to test and I get the `context deadline exceeded` error. I changed the settings from `SSL/TLS port 465` to `STARTTLS and port 587` and I got:
`error sending SMTP test (Test connection): 451 4.4.2 Timeout waiting for data from client`

I was considering reverting back to the previous nginx version, but looking further back in my log file I see more `context deadline exceeded` errors:
```
2024/07/31 09:00:00 manager.go:507: start processing campaign (xxx)
2024/07/31 09:00:05 manager.go:398: error sending message in campaign xxx: subscriber 91798: context deadline exceeded
2024/07/31 09:00:06 manager.go:398: error sending message in campaign xxx: subscriber 27575: context deadline exceeded
2024/07/31 09:00:12 manager.go:398: error sending message in campaign xxx: subscriber 291987: context deadline exceeded
2024/07/31 09:05:19 manager.go:686: campaign (xxx) finished
2024/07/31 16:30:25 subscribers.go:169: error streaming CSV export: write tcp 127.0.0.1:9000->127.0.0.1:11362: write: broken pipe
```

This may have been after I increased the default 500 `settings -> performance -> batch size` to 5000, along with the `settings -> SMTP -> retries` from 2 to 4.

I tried changing the `AUTH protocol` from `LOGIN` to `PLAIN` (saw it here https://github.com/knadh/listmonk/issues/81#issuecomment-631386386) and that worked after some delay. Then I tried it again 15 minutes later and got `451 4.4.2 Timeout waiting for data from client`.

I looked up the errors and found these related issues:

* https://github.com/knadh/listmonk/issues/81
* https://github.com/knadh/listmonk/issues/1624

The first one seems unsolved, and the suggested solution is to retry the messages after some delay. That delay part seems to be a key issue for a number of other errors that people have been experiencing.

One of the participants says they're quite sure it's not a server issue, but looking at my server graph I am seeing some possible downtime issues:
![live graph](https://github.com/user-attachments/assets/87b93308-8a7b-4a60-ad1b-c3fa421993f2)

But my websites running on the server seem to be fine. 

I changed it back to `LOGIN`, lowered the `retries` to 2, and tried again and got `dial tcp <amazon IP>:587: i/o timeout`.

This says it's a firewall issue https://github.com/knadh/listmonk/issues/1504. So I changed it back to TLS, 465 and I got `context deadline exceeded`. I do have 465 and 587 open in my firewall and I haven't changed my firewall, and it worked intermittently, so it seems unlikely to be a firewall issue.

I tried reverting back to the previous nginx version since that's easier than moving servers, and that didn't fix it.

I **created a new Hetzner server** and sent a bunch of test messages without errors. But looking at the graph, the drops with the red line in the SS above appear to be normal. So this is odd.
